src/interfaces/chain.h

```diff
@@ -109,6 +110,10 @@ class Chain
 
         //! Check if transaction will be final given chain height current time.
         virtual bool checkFinalTx(const CTransaction& tx) = 0;
+
+        //! Add transaction to memory pool if the transaction fee is below the
+        //! amount specified by absurd_fee (as a safeguard). */
+        virtual bool submitToMemoryPool(CTransactionRef tx, CAmount absurd_fee, CValidationState& state) = 0;
```

Comment:
in commit d02b34c8a8bd446c9620fe626b4379617f9a9639:

Could take a const ref to `CTransactionRef` to avoid incrementing the shared ptr counter?